### PR TITLE
Use modelopt for fp8 and fp4 by default (#5251)

### PR DESCRIPTION
**Pull request by bcentner**
## Motivation

To use modelopt for fp8 and fp4 by default: https://github.com/sgl-project/sglang/issues/5251

## Modifications

- Uses modelopt as default for fp8
- Adds documentation for quantitization
- Adds test case verifying default

## Checklist

- [x] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [x] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [x] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [x] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [x] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [x] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

File: docs/backend/quantization.md [hunk 1]
```diff
@@ -66,13 +66,45 @@ model.quantize(calibration_dataset, batch_size=2) # quantize
...
 # install
-pip install llmcompressor
+pip install nvidia-modelopt
+```
+
+Here's an example of how to quantize a model to FP8 using ModelOpt:
+
+```python
+from transformers import AutoTokenizer
+from modelopt.torch.quantization import quantize, QuantConfig
```
Comment (Edwardf0t1) [inline docs/backend/quantization.md]: I think `QuantConfig` is from TRT-LLM, not modelopt, were you actually able to import it?

File: docs/backend/quantization.md [hunk 2]
```diff
@@ -66,13 +66,45 @@ model.quantize(calibration_dataset, batch_size=2) # quantize
...
+MODEL_ID = "meta-llama/Meta-Llama-3-8B-Instruct"
+tokenizer = AutoTokenizer.from_pretrained(MODEL_ID)
+
+# Step 2: Configure and apply quantization
+config = QuantConfig(weights_precision="fp8")
+model = quantize(MODEL_ID, config)
+
+# Step 3: Save the quantized model
+SAVE_DIR = MODEL_ID.split("/")[1] + "-FP8-ModelOpt"
+model.save_pretrained(SAVE_DIR)
```
Comment (Edwardf0t1) [inline docs/backend/quantization.md]: We use [unified HF export API](https://github.com/NVIDIA/TensorRT-Model-Optimizer/blob/main/modelopt/torch/export/unified_export_hf.py#L303) to export the quantized model.

File: docs/backend/quantization.md [hunk 3]
```diff
@@ -121,6 +153,15 @@ python3 -m sglang.launch_server \
     --port 30000 --host 0.0.0.0
 ```
 
+For NVIDIA GPUs, we recommend using `modelopt` quantization method for better performance with FP8 and FP4:
+
+```bash
+python3 -m sglang.launch_server \
+    --model-path meta-llama/Meta-Llama-3.1-8B-Instruct \
```
Comment (Edwardf0t1) [inline docs/backend/quantization.md]: We haven't support directly providing the original model in SGLang to quantize, the model here needs to be pre-quantized by modelopt, e.g., [nvidia/Llama-3.1-8B-Instruct-FP8](https://huggingface.co/nvidia/Llama-3.1-8B-Instruct-FP8/tree/main)

Comment (bcentner) [reply docs/backend/quantization.md]: Thank you for clarifying. I removed this section, because the example I added above shows how to use quantized model

File: docs/backend/quantization.md (hunk 2 above)
Comment (bcentner) [reply docs/backend/quantization.md]: I adjusted this to use export_hf_checkpoint

File: docs/backend/quantization.md (hunk 1 above)
Comment (bcentner) [reply docs/backend/quantization.md]: Thanks for calling this out. I updated the import to include the correct module

**Comment by bcentner** [comment]
Thanks for the feedback @Edwardf0t1, please let me know what further changes are needed

File: docs/backend/quantization.md [hunk 4]
```diff
@@ -66,13 +66,49 @@ model.quantize(calibration_dataset, batch_size=2) # quantize
...
+pip install nvidia-modelopt
+```
+
+Here's an example of how to quantize a model to FP8 using ModelOpt:
+
+```python
+from transformers import AutoTokenizer
+from modelopt.torch.quantization import quantize
+from tensorrt_llm.quantization import QuantConfig
+from tensorrt_llm.modelopt.torch.export import export_hf_checkpoint
```
Comment (shimizust) [inline docs/backend/quantization.md:84]: Should this be:
```
from modelopt.torch.export.unified_export_hf import export_hf_checkpoint
```

File: docs/backend/quantization.md [hunk 5]
```diff
@@ -66,13 +66,49 @@ model.quantize(calibration_dataset, batch_size=2) # quantize
...
+
+
+# Step 1: Load the model's tokenizer
+MODEL_ID = "meta-llama/Meta-Llama-3-8B-Instruct"
+tokenizer = AutoTokenizer.from_pretrained(MODEL_ID)
+
+# Step 2: Configure and apply quantization
+config = QuantConfig(weights_precision="fp8")
```
Comment (shimizust) [inline docs/backend/quantization.md:92]: Should we be using modelopt's default config instead?: `modelopt.torch.quantization.FP8_DEFAULT_CFG`?

File: docs/backend/quantization.md [hunk 6]
```diff
@@ -66,13 +66,49 @@ model.quantize(calibration_dataset, batch_size=2) # quantize
...
 
-#### Using [LLM Compressor](https://github.com/vllm-project/llm-compressor/)
+#### Using [ModelOpt](https://github.com/nvidia/modelopt) (Recommended for NVIDIA GPUs)
+
+ModelOpt is now the recommended tool for FP8 and FP4 quantization on NVIDIA GPUs.
 
 ```bash
 # install
-pip install llmcompressor
+pip install nvidia-modelopt
```
Comment (shimizust) [inline docs/backend/quantization.md]: I needed to `pip install nvidia-modelopt[hf,torch]`
